src/app/shared/services/report/report.service.ts

```diff
@@ -27,41 +28,45 @@ export class ReportService {
             responseType: 'arraybuffer'
         }).pipe(
             map(data => {
-                const file = new Blob([data], {type: 'application/pdf'});
                 const fileName = [
                     'UserTransactions',
                     this.datePipe.transform(new Date(), 'yMdhhmmss'),
                     `${!!from ? 'from' + params['from'] : ''}`,
                     `${!!to ? 'to' + params['to'] : ''}`
                 ].filter(text => !!text).join('_') + '.pdf';
 
-                const fileURL = URL.createObjectURL(file);
-                const link = document.createElement('a');
-                link.href = fileURL;
-                link.download = fileName;
-                link.click();
```

Comment:
This chunk was duplicated, so it's convenient to refactor it in a single method.